back-end/src/base/calls/calls.repository.ts

```diff
@@ -49,7 +49,7 @@ export class CallsRepository {
 		const periodIndex = date.getHours() < 16 ? 0 : 1;
 		const call = await this.db
 			.collection('calls')
-			.findOne({ course: courseObjectId, period: period[periodIndex] });
+			.findOne({ course: courseObjectId, period: period[periodIndex], date: this.getDate(date) });
 
 		if (!user) {
 			throw new NotFoundException('User not found');
```

Comment:
I'll switch the classic ErrorException with ServiceError as the try catch is automatic if you throw it.